src/SwapRouter.sol

```diff
@@ -0,0 +1,210 @@
+    }
+
+    /// @dev Direction => True => USDC -> Token
+    /// @dev Direction => False =>  Token -> USDC
+    function estimateAndSwapTokens(
+        bool direction,
+        address token,
+        uint256 amountToSwap,
+        address recipient,
+        uint256 slippage,
+        bytes memory data
+    ) external override returns (uint256 amountOut) {
+        require(
+            token != address(CRV) && token != address(CVX),
+            "SwapRouter :: token"
+        );
+        require(amountToSwap > 0, "SwapRouter :: amountToSwap");
+        require(slippage > 0 && slippage <= 100, "SwapRouter :: slippage");
+
+        IERC20Metadata token0 = direction ? USDC : IERC20Metadata(token);
+        IERC20Metadata token1 = direction ? IERC20Metadata(token) : USDC;
+
+        token0.safeTransferFrom(recipient, address(this), amountToSwap);
+
+        uint256 expectedAmountOut = getTokenPriceInUSD(address(token1));
+
+        _swapTokens(
+            token0,
+            token1,
+            amountToSwap,
+            expectedAmountOut,
+            slippage,
+            data
+        );
+        amountOut = token1.balanceOf(address(this));
+
+        token1.safeTransfer(recipient, amountOut);
+        token0.safeTransfer(recipient, token0.balanceOf(address(this)));
+    }
+
+    /// @dev Direction => True => CRV -> CVXCRV
+    /// @dev Direction => False =>  CVXCRV -> CRV
+    function swapOnCRVCVXCRVPool(
+        bool direction,
+        uint256 amount,
+        address recipient
+    ) external override returns (uint256 amountOut) {
+        IERC20Metadata swapToken = direction ? CRV : CVXCRV;
+        IERC20Metadata recievedToken = direction ? CVXCRV : CRV;
+
+        require(
+            amount > 0 && amount <= swapToken.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        swapToken.safeTransferFrom(msg.sender, address(this), amount);
+        crvcvxcrvPool.exchange(
+            int8(direction ? 0 : 1),
+            int8(direction ? 1 : 0),
+            amount,
+            0,
+            address(this)
+        );
+
+        amountOut = recievedToken.balanceOf(address(this));
+
+        swapToken.safeTransfer(recipient, swapToken.balanceOf(address(this)));
+        recievedToken.safeTransfer(recipient, amountOut);
+    }
+
+    function burn3CRVForUSDC(uint256 amount, address recipient)
+        external
+        override
+        returns (uint256 amountOut)
+    {
+        require(
+            amount > 0 && amount <= _3CRV.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        _3CRV.safeTransferFrom(msg.sender, address(this), amount);
+        /// @dev i = 1 --> USDC
+        _3crvPool.remove_liquidity_one_coin(amount, 1, 0);
+
+        amountOut = USDC.balanceOf(address(this));
```

Comment:
check usdc balance before swap and return the difference

Reply:
There wont be any funds at any point in the contract right?